File: CleaningService.java
import java.util.Date;
import java.text.SimpleDateFormat;
// Object: CleaningService class
public class CleaningService extends Amenity {
    private Date time;

    public CleaningService() {
        super(500);
    }

    public CleaningService(float price) {
        super(price);
    }

    public CleaningService(Date time) {
        super(500);
        this.time = time;
    }
    
    public CleaningService(float price, Date time) {
        super(price);
        this.time = time;
    }

    public Date getTime() {
        return this.time;
    }

    public void setTime(Date time) {
        this.time = time;
    }

    public String getDescription() {
        SimpleDateFormat format = new SimpleDateFormat("hh:mm");
        return "Cleaning service at " + this.getPrice() + " Baht at " + format.format(this.getTime());
    }

    public void makeAppointment()
    {
        System.out.println("What time do you want to get cleaning service? (hh.mm)");
        Date time = Input.getTime();
        this.setTime(time);
    }
}

File: python/CleaningService.py
from datetime import datetime
from Amenity import *

class CleaningService(Amenity):
    def __init__(self):
        super().__init__()
        self.price = 500
        self.__time = datetime(2022, 1, 30, 12, 0, 0)
    
    @property
    def time(self):
        return self.__time
    @time.setter
    def time(self, time):
        self.__time = time
    
    def getDescription(self):
        return "Cleaning amenity at 500 baht per time at " + self.time.strftime("%H:%M:%S")

class CleaningServicePlus(Amenity):
    def __init__(self):
        super().__init__()
        self.price = 600
        self.__time = datetime(0, 0, 0, 12, 0, 0)
    
    @property
    def time(self):
        return self.__time
    @time.setter
    def time(self, time):
        self.__time = time
    
    def getDescription(self):
        return "Cleaning amenity at 600 baht per time at " + self.time.strftime("%H:%M:%S")